akka-stream/src/main/scala/akka/stream/io/OutputStreamSource.scala

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.io
+
+import java.io.OutputStream
+
+import akka.stream._
+import akka.stream.impl.io.OutputStreamSource
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+/**
+ * Source which allows to use [[OutputStream]] to interact with reactive stream.
+ */
+object OutputStreamSource {
+  import scala.language.postfixOps
+
+  final val DefaultChunkSize = 8192
+  final val DefaultAttributes = Attributes.name("OutputStreamSource")
+
+  /**
+   * Creates a synchronous (Java 6 compatible) Source.
+   *
+   * It materializes a [[Future]] containing the number of bytes written to OutputStream upon completion and
+   * [[OutputStream]] to interacting with reactive stream.
+   */
+  def apply(timeout: FiniteDuration = 5.seconds): Source[ByteString, (OutputStream, Future[Long])] = {
+    new Source(new OutputStreamSource(timeout, DefaultAttributes, Source.shape("OutputStreamSource")))
+  }
+  /**
+   * Creates a synchronous (Java 6 compatible) Source.
+   *
+   * It materializes a [[Future]] containing the number of bytes written to OutputStream upon completion and
+   * [[OutputStream]] to interacting with reactive stream.
+   */
+  def create(): javadsl.Source[ByteString, Pair[OutputStream, Future[java.lang.Long]]] =
+    apply().asJava.asInstanceOf[javadsl.Source[ByteString, Pair[OutputStream, Future[java.lang.Long]]]]
```

Comment:
Can you add a test it really works with `j.l.Long`? There's been weird cases when conversion blew up... forgot the exact cases though, better to cover with tests